From 2954a6f12f250890ec2433cec03ba92784d613e8 Mon Sep 17 00:00:00 2001
From: He Ying <heying24@huawei.com>
Date: Tue, 6 Apr 2021 05:42:00 -0400
Subject: [PATCH] firmware: qcom-scm: Fix QCOM_SCM configuration
Git-commit: 2954a6f12f250890ec2433cec03ba92784d613e8
Patch-mainline: v5.13-rc1
References: git-fixes

When CONFIG_QCOM_SCM is y and CONFIG_HAVE_ARM_SMCCC
is not set, compiling errors are encountered as follows:

Drivers/firmware/qcom_scm-smc.o: In function `__scm_smc_do_quirk':
qcom_scm-smc.c:(.text+0x36): undefined reference to `__arm_smccc_smc'
Drivers/firmware/qcom_scm-legacy.o: In function `scm_legacy_call':
qcom_scm-legacy.c:(.text+0xe2): undefined reference to `__arm_smccc_smc'
Drivers/firmware/qcom_scm-legacy.o: In function `scm_legacy_call_atomic':
qcom_scm-legacy.c:(.text+0x1f0): undefined reference to `__arm_smccc_smc'

Note that __arm_smccc_smc is defined when HAVE_ARM_SMCCC is y.
So add dependency on HAVE_ARM_SMCCC in QCOM_SCM configuration.

Fixes: 916f743da354 ("firmware: qcom: scm: Move the scm driver to drivers/firmware")
Reported-by: Hulk Robot <hulkci@huawei.com>
Signed-off-by: He Ying <heying24@huawei.com>
Link: https://lore.kernel.org/r/20210406094200.60952-1-heying24@huawei.com
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/firmware/Kconfig |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/firmware/Kconfig
+++ b/drivers/firmware/Kconfig
@@ -219,6 +219,7 @@ config INTEL_STRATIX10_SERVICE
 config QCOM_SCM
 	bool
 	depends on ARM || ARM64
+	depends on HAVE_ARM_SMCCC
 	select RESET_CONTROLLER
 
 config QCOM_SCM_DOWNLOAD_MODE_DEFAULT
